src/operations/map/MapOperations.ts

```diff
@@ -86,44 +107,97 @@ export class MapOperations {
     });
   }
 
-  static setMapInteractionFunctionality(baseMap: mapboxgl.Map): void {
+  static setMapInteractionFunctionality(
+    { countries, ipcData, selectedMapType }: MapProps,
+    baseMap: mapboxgl.Map
+  ): void {
     let hoveredPolygonId: string | number | undefined;
+    const popup = new mapboxgl.Popup({ closeButton: false, closeOnClick: false });
 
-    baseMap.on('mousemove', 'country-fills', (e) => {
-      if (e.features && e.features.length > 0 && (e.features[0] as unknown as CountryMapData).properties.interactive) {
-        if (hoveredPolygonId) {
-          baseMap.setFeatureState({ source: 'countries', id: hoveredPolygonId }, { hover: false });
-        }
-        hoveredPolygonId = e.features[0].id;
-        if (hoveredPolygonId) {
-          baseMap.setFeatureState({ source: 'countries', id: hoveredPolygonId }, { hover: true });
+    baseMap.on('mousemove', 'country-fills', (e: any) => {
+      const feature = e.features && (e.features[0] as CountryMapData);
+
+      if (hoveredPolygonId && (!feature || hoveredPolygonId !== feature.id)) {
+        baseMap.setFeatureState({ source: 'countries', id: hoveredPolygonId }, { hover: false });
+        hoveredPolygonId = undefined;
+        baseMap.getCanvas().style.cursor = '';
+      }
+
+      if (!feature) {
+        popup.remove();
+        return;
+      }
+
+      baseMap.getCanvas().style.cursor = 'pointer';
+      hoveredPolygonId = feature.id;
+      if (hoveredPolygonId) {
+        baseMap.setFeatureState({ source: 'countries', id: hoveredPolygonId }, { hover: true });
+      }
+
+      switch (selectedMapType) {
+        case 'ipc': {
+          const countryName = feature.properties.adm0_name;
+          const ipcColorMap = IPCMapOperations.generateColorMap(ipcData!, countries!);
+
+          if (!ipcColorMap[countryName]) {
+            popup.remove();
+            return;
+          }
+
+          const { date_of_analysis, analysis_period, ipc_percent, ipc_popnmbr } = IPCMapOperations.findIpcData(
+            countryName,
+            ipcData!
+          );
+          const formattedPopNum = ipc_popnmbr.toLocaleString('en-US', {
+            useGrouping: true,
+          });
+          const title = `${countryName}`;
+          const body = `Date of analysis: ${date_of_analysis} <br> Validity period: ${analysis_period}`;
+          const conclusion = `${formattedPopNum} people in IPC/CH Phase 3 and above <br> (${ipc_percent}% of people in the analyzed areas)`;
+
+          const popupContent = this.generatePopupContent(title, body, conclusion);
+          popup.setLngLat(e.lngLat).setHTML(popupContent).addTo(baseMap);
+          break;
```

Comment:
Can't we use our tooltip here? Maybe align with @ArmanpreetGhotra

Reply:
We can for sure, but I prefer using the mapBox pop up which adaptively adjusts its position based on the mouse's location. e.g if there is no space on the right side of the screen, the popup will automatically appear on the left, and so on.

Reply:
![image](https://github.com/user-attachments/assets/ae743593-18e5-4109-993e-eb3646adaee0)

Reply:
![image](https://github.com/user-attachments/assets/40c3bd3e-0dd9-42cb-b104-444bcd5f11f2)